src/Map/MapRasterizer_P.cpp

```diff
@@ -488,17 +480,144 @@ void OsmAnd::MapRasterizer_P::rasterizePolyline(
                 if (prevCross != 0 || !intersect)
                 {
                     simplifyVertexToDirection(context, pVertex, vertex, tempVertex);
-                    path.moveTo(tempVertex.x, tempVertex.y);
+                    if (shift)
+                    {
+                        originalPoints.push_front(tempVertex);
+                        auto normal = Utilities::computeNormalToLine(pVertex, vertex, dir);
+                        tempVertex += normal * offset;
+                        shiftedPoints.push_front(tempVertex);
+                    }
+                    else
+                    {
+                        path.moveTo(tempVertex.x, tempVertex.y);
+                    }
                 }
                 simplifyVertexToDirection(context, vertex, pVertex, tempVertex);
-                path.lineTo(tempVertex.x, tempVertex.y);
+                if (shift)
+                {
+                    originalPoints.push_front(tempVertex);
+                    auto normal = Utilities::computeNormalToLine(pVertex, vertex, dir);
+                    tempVertex += normal * offset;
+                    shiftedPoints.push_front(tempVertex);
+
+                    // fix corner shifts
+                    if (originalPoints.size() >= kLastPointCnt)
+                    {
+                        // calculation of the angle between two outermost segments of the path.
```

Comment:
instead of comment method could be created

Reply:
Moved to the dedicated function.